src/erlcloud_kinesis_impl.erl

```diff
@@ -113,28 +113,32 @@ request_and_retry(Config, Headers, Body, ShouldDecode, {attempt, Attempt}) ->
            [{<<"content-type">>, <<"application/x-amz-json-1.1">>} | Headers],
            Body, erlcloud_aws:get_timeout(Config), Config) of
 
-        {ok, {{200, _}, _, RespBody}} ->
+        {ok, {{200, _}, RespHeaders, RespBody}} ->
             Result = case ShouldDecode of
                          true  -> decode(RespBody);
                          false -> RespBody
                      end,
             {ok, Result};
 
-        {ok, {{Status, StatusLine}, _, RespBody}} when Status >= 400 andalso Status < 500 ->
+        {ok, {{Status, StatusLine}, RespHeaders, RespBody}} when Status >= 400 andalso Status < 500 ->
+            error_logger:error_msg("Kinesis response headers 400-499 : ~p", [RespHeaders]),
```

Comment:
in spite it's closed, do not log from the library directly make top level app do logging.